### ISSUE TITLE
[import/order] Regression in `type` behavior with `parent` and `sibling`

### ISSUE THREAD

**Issue by erunion** [import-js/eslint-plugin-import#2683]
We have a complex `import/order` config and are seeing some new behavior with our `type` imports on v2.27.5 where previously sorted parent → sibling type imports want to now be sorted as sibling → parent.

```js
'import/order': [
  'error',
  {
    'alphabetize': {
      order: 'asc',
      caseInsensitive: true,
    },
    'groups': ['type', 'builtin', 'external', 'internal', 'parent', 'sibling', 'index', 'object'],
    'newlines-between': 'always',
    'warnOnUnassignedImports': true,
  },
],
```

### Before

```js
import type { CommandOptions } from '../lib/baseCommand';
import type { Options } from './openapi/validate';

import Command from '../lib/baseCommand';
import isHidden from '../lib/decorators/isHidden';

import OpenAPIValidateCommand from './openapi/validate';
```

### After

```js
import type { Options } from './openapi/validate';
import type { CommandOptions } from '../lib/baseCommand'; <-- this line was moved 

import Command from '../lib/baseCommand';
import isHidden from '../lib/decorators/isHidden';

import OpenAPIValidateCommand from './openapi/validate';
```

**Comment by ljharb** [comment]
I think this is a duplicate of #2682?

**Comment by erunion** [comment]
@ljharb I'm not sure because the non-`type` imports are fine here, it's just `import type` that's affected.

**Comment by ljharb** [comment]
cc @Pearce-Ropion in case this is related, but if not, it'd still be great to tackle it in the same PR

**Comment by ljharb** [comment]
`types` is already its own group, and its contents should have the top-level `groups` sorting, minus `types`, applied.

**Comment by Pearce-Ropion** [comment]
Quick progress update, I have a fix for this working locally. I'm working on adding a few more unit test cases to maintain code coverage. I should have a PR up by the end of the week.

**Comment by Pearce-Ropion** [comment]
I've put up #2721. I think it needs more unit testing but figured I'd get the PR out there
